packages/client/hmi-client/src/components/model-template/tera-model-template-editor.vue

```diff
@@ -140,26 +161,42 @@ let currentPortPosition: Position = { x: 0, y: 0 };
 let isMouseOverCanvas = false;
 let canvasTransform = { x: 0, y: 0, k: 1 };
 let isMouseOverPort = false;
+let junctionIdForNewEdge: number | null = null;
 
-const modelTemplates = ref<any[]>([]); // ([{ id: 1, name: 'Template name', x: 300, y: 40 }]);
-const edges = ref<ModelTemplateEdge[]>([]);
+const modelTemplates = ref<any[]>([]);
+const junctions = ref<ModelTemplateJunction[]>([]);
 
 const newModelTemplate = ref();
 const newEdge = ref();
 const isCreatingNewEdge = computed(
 	() => newEdge.value && newEdge.value.points && newEdge.value.points.length === 2
 );
 
+function collisionFn(p: Position) {
+	const buffer = 50;
+	for (let i = 0; i < modelTemplates.value.length; i++) {
+		const checkingNode = modelTemplates.value[i].metadata.templateCard;
+		// FIXME: Thi is  a hack to get around hierarhical geometries, will need to
+		// relax this guard.
+		// if (node.nodes && node.nodes.length > 0) continue;
```

Comment:
Let's remove the copy-pasta comments - we don't have nested hierarchies, at least not in a way that the edges will need to cross nesting levels.